doc/Programming-guide/Using-Kaa-endpoint-SDKs/C++/SDK-RPi/index.md

```diff
@@ -5,4 +5,107 @@ permalink: /:path/
+        tar -xvzf zlib-1.2.8.tar.gz
+        cd zlib-1.2.8
+        ./configure --prefix=${INSTALLDIR}/arm-linux-gnueabihf/libc/usr
+        make && make install
+
+3. Download and install bzib2.
+
+        cd $RPI_ROOT
+        wget http://www.bzip.org/1.0.6/bzip2-1.0.6.tar.gz
+        tar -xvzf bzip2-1.0.6.tar.gz
+        cd bzip2-1.0.6
+        sed -e "/^all:/s/ test//" -i Makefile
+        make CC="${CC}" AR="${AR}"
+        make PREFIX=${INSTALLDIR}/arm-linux-gnueabihf/libc/usr install
+
+4. Download and install boost.
+
+        cd $RPI_ROOT
+        wget -O boost_1_59_0.tar.bz2 http://sourceforge.net/projects/boost/files/boost/1.59.0/boost_1_59_0.tar.bz2/download
+        tar -xvjf boost_1_59_0.tar.bz2
+        cd boost_1_59_0
+        ./bootstrap.sh
+
+    Edit the `project-config.jam` file. Add 'using gcc : arm : arm-linux-gnueabihf-c++ ;' instead of  'using gcc ;':
+
+        ./bjam install toolset=gcc-arm --prefix=${INSTALLDIR}/arm-linux-gnueabihf/libc/usr
+
+5. Install Avro for target and host. Also Avro depends on some Boost components. So they need to be installed too.
+
+    Install Avro for host:
+
+        cd $RPI_ROOT
+        sudo apt-get install libboost-dev libboost-filesystem-dev libboost-iostreams-dev libboost-program-options-dev libboost-system-dev
+        wget https://archive.apache.org/dist/avro/avro-1.7.5/cpp/avro-cpp-1.7.5.tar.gz
+        mkdir -p avro/host
+        tar -xvzf avro-cpp-1.7.5.tar.gz -C ./avro/host
+        unset CC
+        cd ./avro/host/avro-cpp-1.7.5
+        mkdir build && cd build
+
+        cmake ..
+        sudo make install
+
+    Build Avro for target (Avro should be patched to be built):
+
+        cd $RPI_ROOT
+        mkdir -p avro/target
+        tar -xvzf avro-cpp-1.7.5.tar.gz -C ./avro/target
+        cd ./avro/target/avro-cpp-1.7.5
+
+        wget https://raw.githubusercontent.com/kaaproject/kaa/8e65212dfb855363e1a8977d4053041c80d785c7/client/client-multi/client-c/toolchains/rpi.cmake
+        wget https://raw.githubusercontent.com/kaaproject/kaa/967970ec57fb0a62c23ffe573385bf0d0299d977/client/client-multi/client-cpp/tools/avro-cpp-disable-tests.patch
+        patch < avro-cpp-disable-tests.patch
+        mkdir build && cd build
+        export CC=${CROSS}-gcc
+        cmake -DCMAKE_INSTALL_PREFIX:PATH=${INSTALLDIR}/arm-linux-gnueabihf/libc/usr -DCMAKE_TOOLCHAIN_FILE=../rpi.cmake ..
+        make && make install
+
+6. Build Botan.
+
+        cd $RPI_ROOT
+        wget http://botan.randombit.net/releases/Botan-1.11.27.tgz
+        tar -xvzf Botan-1.11.27.tgz
+        cd Botan-1.11.27
+        python configure.py --cpu=arm --cc-bin=${CROSS}-g++ --prefix=${INSTALLDIR}/arm-linux-gnueabihf/libc/usr
+        make && make install
+        cp -r ${INSTALLDIR}/arm-linux-gnueabihf/libc/usr/include/botan-1.11/botan/ ${INSTALLDIR}/arm-linux-gnueabihf/libc/usr/include
+        rm -r ${INSTALLDIR}/arm-linux-gnueabihf/libc/usr/include/botan-1.11
+
+7. [Generate C++ SDK](#TODO).
```

Comment:
C++ SDK generation page is present in 
Administration-guide/Tenants-and-applications-management/#generating-endpoint-sdk